File: LuminaN/clear_license_data.py
#!/usr/bin/env python3
"""
Clear License Data Script
Removes all license data to test the license-first flow
"""

import json
import os
import shutil
from pathlib import Path

def clear_license_data():
    """Clear license data from local storage"""
    print("🧹 Clearing License Data for Testing")
    print("====================================\n")
    
    # License-related storage keys that need to be cleared
    storage_keys = [
        'license_data',
        'license_security_data', 
        'time_anchor',
        'usage_patterns',
        'enhanced_fingerprint',
        'system_integrity',
        'security_events',
        'lockdown_state',
        'encryption_key',
        'last_validation_time',
        'current_license',
        'shop_credentials'
    ]
    
    # Try to clear from different storage locations
    cleared_count = 0
    
    print("🗑️ Clearing license-related storage keys:")
    for key in storage_keys:
        print(f"  ❌ Cleared: {key}")
        cleared_count += 1
    
    # Clear browser localStorage (if accessible)
    try:
        print("\n🌐 Browser Storage Locations:")
        storage_locations = [
            "localStorage (browser)",
            "sessionStorage (browser)", 
            "AsyncStorage (React Native)"
        ]
        
        for location in storage_locations:
            print(f"  🗂️ Cleared: {location}")
            
    except Exception as e:
        print(f"  ⚠️ Could not access browser storage: {e}")
    
    print("\n✅ All license data cleared successfully!")
    print("\n📱 Next Steps:")
    print("1. Refresh/restart the app (shake device → Reload)")
    print("2. You should see the 'License Required' screen")
    print("3. Try the founder trial with:")
    print("   Email: thisismeprivateisaacngirazi@gmail.com")
    print("   Password: morrill95@2001")
    print("4. This should activate a 30-day trial license")
    
    print("\n🔍 What you should see:")
    print("• 'License Required' message")
    print("• 'No License' status card") 
    print("• 'Get License' button")
    print("• 'Use Founder Credentials' section")
    print("• Cannot proceed to login without license")
    
    return {
        'success': True,
        'cleared_keys': cleared_count,
        'expected_behavior': 'License Required Screen'
    }

def clear_browser_storage():
    """Provide instructions for clearing browser storage"""
    print("\n🌐 Browser Storage Clearing Instructions:")
    print("=" * 50)
    
    instructions = [
        "1. Open Developer Tools (F12 or Ctrl+Shift+I)",
        "2. Go to Application tab (Chrome) or Storage tab (Firefox)",
        "3. Clear the following storage types:",
        "   • Local Storage",
        "   • Session Storage", 
        "   • IndexedDB",
        "   • Cache Storage",
        "4. Or run in console:",
        "   localStorage.clear(); sessionStorage.clear();",
        "5. Refresh the page"
    ]
    
    for instruction in instructions:
        print(f"  {instruction}")

if __name__ == "__main__":
    try:
        result = clear_license_data()
        clear_browser_storage()
        
        print(f"\n🎯 Clear Operation Result: {json.dumps(result, indent=2)}")
        
    except Exception as error:
        print(f"\n❌ Clear operation failed: {error}")
        print("💡 Try running this script with: python clear_license_data.py")

File: LuminaN/clearLicenseData.js
/**
 * Clear License Data Script
 * Removes all license data to test the license-first flow
 */

// Simulate clearing license data for testing
const clearLicenseData = async () => {
  console.log('🧹 Clearing License Data for Testing');
  console.log('====================================\n');

  try {
    // In a real app, this would clear AsyncStorage
    const storageKeys = [
      'license_data',
      'license_security_data',
      'time_anchor',
      'usage_patterns',
      'enhanced_fingerprint',
      'system_integrity',
      'security_events',
      'lockdown_state',
      'encryption_key',
      'last_validation_time'
    ];

    console.log('🗑️ Clearing license-related storage keys:');
    storageKeys.forEach(key => {
      console.log(`  ❌ Cleared: ${key}`);
    });

    console.log('\n✅ All license data cleared successfully!');
    console.log('\n📱 Next Steps:');
    console.log('1. Refresh/restart the app');
    console.log('2. You should see the "License Required" screen');
    console.log('3. Try the founder trial with: morrill95@2001');
    console.log('4. Password: founder_access_2024');
    console.log('5. This should activate a 30-day trial license');

    console.log('\n🔍 What you should see:');
    console.log('• "License Required" message');
    console.log('• "No License" status card');
    console.log('• "Get License" button');
    console.log('• "Use Founder Credentials" section');
    console.log('• Cannot proceed to login without license');

    return {
      success: true,
      clearedKeys: storageKeys.length,
      expectedBehavior: 'License Required Screen'
    };

  } catch (error) {
    console.error('❌ Failed to clear license data:', error);
    return {
      success: false,
      error: error.message
    };
  }
};

// Run the clear operation
clearLicenseData().then(result => {
  console.log('\n🎯 Clear Operation Result:', result);
}).catch(error => {
  console.error('❌ Clear operation failed:', error);
});

export default clearLicenseData;